packages/lit-html/src/directive-helpers.ts

```diff
@@ -42,17 +48,35 @@ export const TemplateResultType = {
 export type TemplateResultType =
   (typeof TemplateResultType)[keyof typeof TemplateResultType];
 
+type isTemplateResultType = {
```

Comment:
Capitalize all type names, and I wouldn't include "Type" in the name. I think `IsTemplateResult` is fine.